corehq/apps/repeaters/models.py

```diff
@@ -301,7 +283,15 @@ def get_auth(self):
     def fire_for_record(self, repeat_record):
         headers = self.get_headers(repeat_record)
         auth = self.get_auth()
-        payload = self.get_payload_and_handle_exception(repeat_record)
+
+        try:
+            payload = self.get_payload(repeat_record)
+        except Exception as e:
+            # todo: seems like this just does everything that handle_exception does but not as well.
+            # todo:   seems like they could be combined
```

Comment:
OK, I'm not sure I follow (this calls `raise` right below `handle_payload_exception`, maybe there was a typo in what you wrote or I'm misunderstanding). I'm also leaning towards leaving this like this with the todo comment and addressing later, since it feels like anything that addresses it is likely to (possibly significantly) change behavior. I imagine this `except` block is mostly here to catch errors in custom repeaters' get_payload call, which may be more complicated than the built-ins which just fetch a case/form